core/src/elo-connector/elo-connector.service.ts

```diff
@@ -0,0 +1,291 @@
+     * @param winningTeam The team that should win the NCP. Required if seriesType=Fixture and isNcp=true.
+     * @param numReplays The number of replays that should be in the series. Optional. Used to add dummy replays in place of replays that weren't submitted for some reason.
+     * @returns A string containing a summary of the actions that took place when the processing has completed.
+     */
+    async markSeriesNcp(seriesId: number, isNcp: boolean, seriesType: SeriesType, winningTeamInput?: Team, numReplays?: number): Promise<string> {
+        const r = Math.floor(Math.random() * 10000);
+        this.logger.verbose(`(${r}) begin markSeriesNcp: seriesId=${seriesId}, isNcp=${isNcp}, winningTeam=${winningTeamInput}`);
+
+        // Find the winning team and it's franchise profile, since that's where
+        // team names are in Sprocket.
+        let winningTeam = await this.teamRepository.findOne(winningTeamInput?.id, {where: {id: winningTeamInput?.id}, relations: ["franchise", "franchise.profile"] });
+
+        // const series = await this.ss.getSeriesById(seriesId, ["fixture.home",
+        // "fixture.away"]);
+        const series = await this.matchRepository.findOneOrFail(seriesId, {where: {id: seriesId}, relations: ["matchParent", "rounds"] });
+
+        if (seriesType === SeriesType.Fixture) {
+
+            if (!series.matchParent.fixture) {
+                return `Error: series with id=\`${seriesId}\` has no associated fixture. If this is a scrim, please use \`!ncpScrim\` or \`!unNcpScrim\`. If this actually is a league play series, ping Bot Team.`;
+            }
+
+            // Winning team must be specified if NCPing replays
+            if (isNcp && !winningTeam) {
+                return "When NCPing a series associated with a fixture, you must specify a winningTeam";
+            }
+
+            // Check to make sure the winning team played in the series/fixture
+            if (winningTeam
+                && series.matchParent.fixture.homeFranchise !== winningTeam.franchise
+                && series.matchParent.fixture.awayFranchise !== winningTeam.franchise) {
+                return `The team \`${winningTeam?.franchise.profile.title}\` did not play in series with id \`${series.id}\` (${series.matchParent.fixture.awayFranchise.profile.title} v. ${series.matchParent.fixture.homeFranchise.profile.title}), and therefore cannot be marked as the winner of this NCP. Cancelling process with no action taken.`;
+            }
+
+        } else if (seriesType === SeriesType.Scrim) {
+            if (!series.matchParent.scrimMeta) {
+                return `Error: series with id=\`${seriesId}\` has no associated scrim. If this is a league play series, please use \`!ncpSeries\` or \`!unNcpSeries\`. If this actually is a scrim, ping Bot Team.`;
+            }
+            // winningTeam = await this.ts.resolveTeam("FA");
+            winningTeam = await this.teamRepository.findOneOrFail(0); // TODO: ID for FA team
```

Comment:
We _should_ be able to pull the FA team off of the scrim, right? Otherwise, does this TODO need to be addressed in this PR, or in another ticket?

Reply:
I left it this way for now so I could focus on the core logic of NCPs. With this current code, we'll be able to squeak by with a dummy team in the DB that lets us know its FA. 

I do want to revisit this (and some of your other comments re: directly injecting repositories vs splitting out a service) in a future effort. Whether we want that as part of this PR or not is up to you. I'd prefer to write that up as part of a refactor ticket I can do once MVP goes live.

Reply:
https://app.clickup.com/t/2pdz0e8